tools/clang/test/HLSLFileCheck/validation/ResourceResolvingNoDI.hlsl

```diff
@@ -0,0 +1,25 @@
+// RUN: %dxc -T cs_6_0 -fdisable-debug-info %s 2>&1 | FileCheck %s
+
+// CHECK: Function: main: note: Debug information is disabled which may impact diagnostic location accuracy. Re-run without -fdisable-debug-info to improve accuracy.
+// CHECK-NOT: note: Debug information is disabled
+
+RWBuffer<int> Buf[2];
+RWBuffer<int> Out;
+
+int getVal(bool B, int Idx) {
+  RWBuffer<int> Local;
+  if (B) Local = Buf[Idx];
+  return Local[Idx];
+}
+
+int getValASecondTime(bool B, int Idx) {
+  RWBuffer<int> Local;
+  if (B) Local = Buf[Idx];
+  return Local[Idx];
+}
+
+[numthreads(1,1,1)]
+void main(uint GI : SV_GroupIndex) {
+  Out[GI] = getVal(false, GI);
+  Out[GI+1] = getValASecondTime(false, GI);
+}
```

Comment:
A test that verifies that line info isn't available might be useful. You can look at the tests Adam changed in #3606 for some examples.

Reply:
The note this test checks for is only emitted if line information isn't available... does that not cover that case or are you wanting me to verify that the IR has no line information before we get into the backend?